File: .luna/coderailflow/heal-screenshot.js
const { chromium } = require('playwright');

const BASE = 'http://localhost:5173';
const ROUTES = [
  { path: '/', name: 'landing' },
  { path: '/app', name: 'dashboard' },
  { path: '/billing', name: 'billing' },
  { path: '/projects', name: 'projects' },
];

const VIEWPORTS = [
  { width: 375, height: 812, name: 'mobile' },
  { width: 768, height: 1024, name: 'tablet' },
  { width: 1440, height: 900, name: 'desktop' },
];

(async () => {
  const browser = await chromium.launch({ headless: true });
  const outDir = process.argv[2] || '.luna/coderailflow/heal-report/iteration-1/screenshots';

  for (const route of ROUTES) {
    for (const vp of VIEWPORTS) {
      const page = await browser.newPage({ viewport: { width: vp.width, height: vp.height } });
      try {
        await page.goto(`${BASE}${route.path}`, { waitUntil: 'networkidle', timeout: 10000 });
        await page.waitForTimeout(1000);
        const filename = `${outDir}/${route.name}-${vp.name}.png`;
        await page.screenshot({ path: filename, fullPage: true });
        console.log(`OK: ${filename}`);
      } catch (e) {
        console.error(`FAIL: ${route.name}-${vp.name}: ${e.message}`);
      }
      await page.close();
    }
  }
  await browser.close();
  console.log('Done');
})();

File: .luna/coderailflow/heal-screenshot.py
import sys, os
from playwright.sync_api import sync_playwright

BASE = 'http://localhost:5173'
ROUTES = [
    ('/', 'landing'),
    ('/app', 'dashboard'),
    ('/billing', 'billing'),
    ('/projects', 'projects'),
]
VIEWPORTS = [
    (375, 812, 'mobile'),
    (768, 1024, 'tablet'),
    (1440, 900, 'desktop'),
]

out_dir = sys.argv[1] if len(sys.argv) > 1 else '.luna/coderailflow/heal-report/iteration-1/screenshots'
os.makedirs(out_dir, exist_ok=True)

with sync_playwright() as p:
    browser = p.chromium.launch(headless=True)
    for path, name in ROUTES:
        for w, h, vp_name in VIEWPORTS:
            page = browser.new_page(viewport={'width': w, 'height': h})
            try:
                page.goto(f'{BASE}{path}', wait_until='networkidle', timeout=10000)
                page.wait_for_timeout(1000)
                filename = f'{out_dir}/{name}-{vp_name}.png'
                page.screenshot(path=filename, full_page=True)
                print(f'OK: {filename}')
            except Exception as e:
                print(f'FAIL: {name}-{vp_name}: {e}')
            page.close()
    browser.close()
print('Done')
